### PR TITLE
sim: Suppress "has no symbols" warnings for macOS

### PR DESCRIPTION
**Pull request by yamt**
## Summary
https://github.com/apache/incubator-nuttx/issues/920

### CODE REVIEW

File: binfmt/Makefile [hunk 1]
```diff
@@ -47,6 +47,7 @@ BINFMT_ASRCS  =
 BINFMT_CSRCS  = binfmt_globals.c binfmt_initialize.c binfmt_register.c binfmt_unregister.c
 BINFMT_CSRCS += binfmt_loadmodule.c binfmt_unloadmodule.c binfmt_execmodule.c
 BINFMT_CSRCS += binfmt_exec.c binfmt_copyargv.c binfmt_dumpmodule.c
+BINFMT_CSRCS += dummy.c
```
Comment (xiaoxiang781216) [inline binfmt/Makefile:50]: We hit the similar issue with ceva toolchain before, and find a general solution:
# cofflib -r will core dump if the library contain an object file
# without any public symbol, so let's always generate a dummy global
# function to overcome this issue.
ARCHCPUFLAGS += -DCKSUM_SELF=$(if $<,$(word 1,$(shell cksum $<)))
ARCHCPUFLAGS += -include dummy.h
Here is two related files:
https://github.com/FishsemiCode/nuttx/blob/song-u1/arch/ceva/src/tl4/Toolchain.defs
https://github.com/FishsemiCode/nuttx/blob/song-u1/arch/ceva/src/tl4/dummy.h
Without a general solution, we have to fix the same issue again and again.

Comment (yamt) [reply binfmt/Makefile:50]: my impression is that it's a problem in nuttx, not toolchain.
(well, it should not core dump though)

Comment (xiaoxiang781216) [reply binfmt/Makefile:50]: Anyway, we need a general(no ad-hoc) solution regardless it's a toolchian or nuttx problem.

Comment (acassis) [reply binfmt/Makefile:50]: @yamt and @xiaoxiang781216 what is the consensus? Should we merge it or wait for a more general solution?

Comment (xiaoxiang781216) [reply binfmt/Makefile:50]: @yamt and @acassis here is the PR I suggest: https://github.com/apache/incubator-nuttx/pull/933.

Comment (xiaoxiang781216) [reply binfmt/Makefile:50]: The check result show that #933 work as expect:
https://github.com/apache/incubator-nuttx/pull/936/checks?check_run_id=636779093

Comment (patacongo) [reply binfmt/Makefile:50]: @xiaoxiang781216 Does PR #936  replace this one?  I merged that one.  Or should this one be merged too?

Comment (xiaoxiang781216) [reply binfmt/Makefile:50]: No, we just need merge one.

Comment (yamt) [reply binfmt/Makefile:50]: i feel we merge patches too quickly. can we wait for at least one day when a patch is controversial?

the merged patch is IMO too much hack. it even pollutes the end binary.
if you don't like my approach in this patch, you can just remove the warnings in the wrapper script instead.

Comment (xiaoxiang781216) [reply binfmt/Makefile:50]: Yes, it's too quick. How about you reopen PR and revert my patch with more better fix?

**Pull request closed without merging**